pkg/agent/controller/bgp/controller.go

```diff
@@ -0,0 +1,954 @@
+
+	serviceInformer     cache.SharedIndexInformer
+	serviceLister       corelisters.ServiceLister
+	serviceListerSynced cache.InformerSynced
+
+	egressInformer     cache.SharedIndexInformer
+	egressLister       crdlistersv1b1.EgressLister
+	egressListerSynced cache.InformerSynced
+
+	bgpPolicyInformer     cache.SharedIndexInformer
+	bgpPolicyLister       crdlistersv1a1.BGPPolicyLister
+	bgpPolicyListerSynced cache.InformerSynced
+
+	endpointSliceInformer     cache.SharedIndexInformer
+	endpointSliceLister       discoverylisters.EndpointSliceLister
+	endpointSliceListerSynced cache.InformerSynced
+
+	bgpPolicyState *bgpPolicyState
+
+	k8sClient             kubernetes.Interface
+	bgpPeerPasswords      map[string]string
+	bgpPeerPasswordsMutex sync.RWMutex
+
+	nodeName     string
+	enabledIPv4  bool
+	enabledIPv6  bool
+	podIPv4CIDR  string
+	podIPv6CIDR  string
+	nodeIPv4Addr string
+
+	egressEnabled bool
+
+	newBGPServerFn func(globalConfig *bgp.GlobalConfig) bgp.Interface
+
+	queue workqueue.RateLimitingInterface
+}
+
+func NewBGPPolicyController(ctx context.Context,
+	nodeInformer coreinformers.NodeInformer,
+	serviceInformer coreinformers.ServiceInformer,
+	egressInformer crdinformersv1b1.EgressInformer,
+	bgpPolicyInformer crdinformersv1a1.BGPPolicyInformer,
+	endpointSliceInformer discoveryinformers.EndpointSliceInformer,
+	k8sClient kubernetes.Interface,
+	nodeConfig *config.NodeConfig,
+	networkConfig *config.NetworkConfig) (*Controller, error) {
+	c := &Controller{
+		ctx:                       ctx,
+		nodeInformer:              nodeInformer.Informer(),
+		nodeLister:                nodeInformer.Lister(),
+		nodeListerSynced:          nodeInformer.Informer().HasSynced,
+		serviceInformer:           serviceInformer.Informer(),
+		serviceLister:             serviceInformer.Lister(),
+		serviceListerSynced:       serviceInformer.Informer().HasSynced,
+		egressInformer:            egressInformer.Informer(),
+		egressLister:              egressInformer.Lister(),
+		egressListerSynced:        egressInformer.Informer().HasSynced,
+		bgpPolicyInformer:         bgpPolicyInformer.Informer(),
+		bgpPolicyLister:           bgpPolicyInformer.Lister(),
+		bgpPolicyListerSynced:     bgpPolicyInformer.Informer().HasSynced,
+		endpointSliceInformer:     endpointSliceInformer.Informer(),
+		endpointSliceLister:       endpointSliceInformer.Lister(),
+		endpointSliceListerSynced: endpointSliceInformer.Informer().HasSynced,
+		k8sClient:                 k8sClient,
+		bgpPeerPasswords:          make(map[string]string),
+		nodeName:                  nodeConfig.Name,
+		enabledIPv4:               networkConfig.IPv4Enabled,
+		enabledIPv6:               networkConfig.IPv6Enabled,
+		podIPv4CIDR:               nodeConfig.PodIPv4CIDR.String(),
+		podIPv6CIDR:               nodeConfig.PodIPv6CIDR.String(),
+		nodeIPv4Addr:              nodeConfig.NodeIPv4Addr.IP.String(),
+		egressEnabled:             features.DefaultFeatureGate.Enabled(features.Egress),
```

Comment:
It should use the `o.enableEgress` consistently in case we change how to determine Egress is enabled in the future.

Reply:
Done